File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    
    map<string,int> wMap;
    
    for(int i = 0; i < want.size(); i++)
    {
        wMap[want[i]] += number[i];
    }
    
    for(int i = 0; i <= discount.size() - 10; i++)
    {   
        map<string, int> tMap = wMap; 
        for (int j = 0; j < 10; j++) {
            string item = discount[i + j];
            if (tMap.find(item) != tMap.end()) {
                tMap[item]--;
            }
        }
        
        bool check = true;
        for(auto it : tMap)
        {
            if(it.second > 0)
            {
                check = false;
                break;
            }
        }
        if(check)
        {
            answer++;
        }
    }
    
    
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(string[] want, int[] number, string[] discount) {
        int answer = 0;
        
        Dictionary<string, int> wMap = new Dictionary<string, int>();
        for (int i = 0; i < want.Length; i++)
        {
            wMap[want[i]] = number[i];
        }
        
        for (int i = 0; i <= discount.Length - 10; i++)
        {
            Dictionary<string, int> tMap = new Dictionary<string, int>(wMap);
            
            for (int j = 0; j < 10; j++)
            {
                string item = discount[i+j];
                if(tMap.ContainsKey(item)){
                    tMap[item]--;
                }
            }
            
            bool check = true;
            foreach (var pair in tMap)
            {
                if(pair.Value > 0)
                {
                    check = false;
                    break;
                }
            }
            
            if (check) answer++;
            
        }
        
        return answer;
    }
}